src/connector/src/parser/postgres.rs

```diff
@@ -382,57 +349,166 @@ pub fn postgres_row_to_owned_row(row: tokio_postgres::Row, schema: &Schema) -> O
+                            _ => {
+                                handle_list_data_type!(row, i, name, String, builder);
+                            }
+                        }
+                    }
+                    DataType::Time => {
+                        handle_list_data_type!(row, i, name, chrono::NaiveTime, builder, Time);
+                    }
+                    DataType::Timestamp => {
+                        handle_list_data_type!(
+                            row,
+                            i,
+                            name,
+                            chrono::NaiveDateTime,
+                            builder,
+                            Timestamp
+                        );
+                    }
+                    DataType::Timestamptz => {
+                        handle_list_data_type!(
+                            row,
+                            i,
+                            name,
+                            chrono::DateTime<Utc>,
+                            builder,
+                            Timestamptz
+                        );
+                    }
+                    DataType::Interval => {
+                        handle_list_data_type!(row, i, name, Interval, builder);
+                    }
+                    DataType::Jsonb => {
+                        handle_list_data_type!(row, i, name, serde_json::Value, builder, JsonbVal);
+                    }
+                    DataType::Bytea => {
+                        let res = row.try_get::<_, Option<Vec<Option<Vec<u8>>>>>(i);
+                        match res {
+                            Ok(val) => {
+                                if let Some(v) = val {
+                                    v.into_iter().for_each(|val| {
+                                        builder.append(
+                                            val.map(|v| ScalarImpl::from(v.into_boxed_slice())),
+                                        )
+                                    })
+                                }
+                            }
+                            Err(err) => {
+                                log_error!(name, err, "parse column failed");
+                            }
+                        }
+                    }
+                    DataType::Int256 => {
+                        let res = row.try_get::<_, Option<Vec<Option<PgNumeric>>>>(i);
+                        match res {
+                            Ok(val) => {
+                                if let Some(v) = val {
+                                    // In Debezium, when there's inf, -inf, nan or invalid item in a list, the whole list will fallback null.
+                                    // In Json Parser, when there's a numeric with decimal part in a numeric list, the whole list will fallback null.
+                                    // So we directly return None here to keep backfill behavior consistent with Debezium behavior.
```

Comment:
I am almost lost. Can we avoid adding these bug-compatible logic at 3 different places?

Reply:
Done